auth/basic/db.go

```diff
@@ -41,6 +46,12 @@ type HashedPassword interface {
 	_private() // prevent external implementations
 }
 
+type ValidatedPassword interface {
+	Hash() string
```

Comment:
Have this return the user ID it was validated against rather than the hash. We can use that to ensure the ValidatedPassword is for the correct user in `UpdateTx`